### ISSUE TITLE
@storybook/client-api:clientApi.getStorybook was removed in storyStoreV7

### ISSUE THREAD

**Issue by JBallin** [screener-io/screener-storybook#147]
Both static and non-static builds fail with the error in title (full output below) when using `storyStoreV7`.

## What is storyStoreV7?
Opt-in (requires v6.4+, will likely be default in V7) Storybook feature that improves startup performance.
* Docs
  * [Configure](https://storybook.js.org/docs/react/configure/overview#on-demand-story-loading)
  * [Webpack](https://storybook.js.org/docs/react/builders/webpack#code-splitting)
* [Migration Guide](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#story-store-v7)
* [Announcement blog](https://storybook.js.org/blog/storybook-on-demand-architecture)

## Notes
I believe it's failing on [src/storybook.js#L114](https://github.com/screener-io/screener-storybook/blob/f6589ea71668aa7c4b64441f2e750d39b65cecd4/src/storybook.js#L114).

<details>
<summary>Error Output</summary>

```
DEBUG: previewRoute /iframe.html
DEBUG: GET http://localhost:2000/iframe.html
DEBUG: getStorybook 0
DEBUG: getStorybook Error: Evaluation failed: Error: @storybook/client-api:clientApi.getStorybook was removed in storyStoreV7.
    at http://localhost:2000/vendors-node_modules_REDACTED.iframe.bundle.js:241392:11
    at __puppeteer_evaluation_script__:1:8
    at ExecutionContext.evaluateHandle (/Users/REDACTED/node_modules/puppeteer/lib/ExecutionContext.js:80:15)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async ExecutionContext.evaluate (/Users/REDACTED/node_modules/puppeteer/lib/ExecutionContext.js:48:20)
  -- ASYNC --
    at ExecutionContext.<anonymous> (/Users/REDACTED/node_modules/puppeteer/lib/helper.js:110:27)
    at ExecutionContext.evaluate (/Users/REDACTED/node_modules/puppeteer/lib/ExecutionContext.js:48:31)
    at ExecutionContext.<anonymous> (/Users/REDACTED/node_modules/puppeteer/lib/helper.js:111:23)
    at DOMWorld.evaluate (/Users/REDACTED/node_modules/puppeteer/lib/DOMWorld.js:112:20)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
  -- ASYNC --
    at Frame.<anonymous> (/Users/REDACTED/node_modules/puppeteer/lib/helper.js:110:27)
    at Page.evaluate (/Users/REDACTED/node_modules/puppeteer/lib/Page.js:782:43)
    at Page.<anonymous> (/Users/REDACTED/node_modules/puppeteer/lib/helper.js:111:23)
    at getStorybook (/Users/REDACTED/node_modules/screener-storybook/src/storybook.js:42:15)
    at /Users/REDACTED/node_modules/screener-storybook/src/storybook.js:114:20
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
DEBUG: config.storybookPort 2000
Error getting Storybook object
Please send debug output to help@saucelabs.com
DEBUG: Error: Storybook object not found
    at Object.exports.get [as getStorybook] (/Users/REDACTED/node_modules/screener-storybook/src/storybook.js:312:11)
    at /Users/REDACTED/node_modules/screener-storybook/src/cli.js:47:40
    at tryCatcher (/Users/REDACTED/node_modules/screener-storybook/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/Users/REDACTED/node_modules/screener-storybook/node_modules/bluebird/js/release/promise.js:510:31)
    at Promise._settlePromise (/Users/REDACTED/node_modules/screener-storybook/node_modules/bluebird/js/release/promise.js:567:18)
    at Promise._settlePromise0 (/Users/REDACTED/node_modules/screener-storybook/node_modules/bluebird/js/release/promise.js:612:10)
    at Promise._settlePromises (/Users/REDACTED/node_modules/screener-storybook/node_modules/bluebird/js/release/promise.js:691:18)
    at Promise._fulfill (/Users/REDACTED/node_modules/screener-storybook/node_modules/bluebird/js/release/promise.js:636:18)
    at /Users/REDACTED/node_modules/screener-storybook/node_modules/bluebird/js/release/nodeback.js:42:21
    at /Users/REDACTED/node_modules/screener-storybook/src/storybook.js:299:7
    at done (/Users/REDACTED/node_modules/screener-storybook/src/storybook.js:99:11)
---
Exiting Screener Storybook
```

</details>

**Comment by igoldny** [comment]
yes, same thing happens to me as well.
@loyalchow please help, would be great adding this improvement...

**Comment by loyalchow** [comment]
@igoldny I left Sauce at the end of '21. But I'll reach out, team is probably looking into this

**Comment by igoldny** [comment]
> @igoldny I left Sauce at the end of '21. But I'll reach out, team is probably looking into this

ohh sorry, thnx for looking at it. 🙏

**Comment by swese44** [comment]
Can we please get a status update on this? Thank you!

**Comment by swese44** [comment]
Hi @loyalchow. Sorry to bother you but would you be able to @mention whoever now maintains this project? Thanks!

**Comment by swese44** [comment]
@vrunoa or @xwa130saucer are you able to address this? Thank you!

**Comment by vrunoa** [comment]
@swese44 sorry, Im no longer part of Sauce. @xwa130saucer @sauce-tartar may be able to follow up on this.

**Comment by sauce-tartar** [comment]
@swese44 Hi, I can give you a quick update and I've messaged our SauceLabs Visual team's Senior Product and our Director that you've been looking for help.  Priority it high for it, multiple customers impacted.

We presently have a Spike open on our board about it VSLT-2904.  We do know we need to upgrade the @storybook/client-api and @storybook/store.

**Comment by swese44** [comment]
Thanks @sauce-tartar!

**Comment by swese44** [comment]
Hi @sauce-tartar. It's been another month, is there any update here? The lack of support and updates for a product we pay for us a bit frustrating. Thank you!

**Comment by dswiz77** [comment]
@swese44 this is something the team has been on for the last few weeks so I think it would be best if we can sync next week to update you and work towards a resolution.  We will reach out to you directly.

**Comment by swese44** [comment]
Sounds good, thanks @dswiz77!

**Comment by sauce-tartar** [comment]
Happy to share some good news for `storyStoreV7`.  It's here as an alpha program, rolling along with presently with this PR.  Having passed a decent sized CI matrix, we've been able to meet a good deal of backward compatibility however have bumped it to 1.0.0 given the extent of the internals not to mention a major Puppeteer update.

https://github.com/screener-io/screener-storybook/pull/157

If you have have chance please give this a run.  One major known issue is Screener Steps broken but we have that at hand as well coming soon.

```
screener-storybook@alpha
```

**Comment by igoldny** [comment]
Great news! Thank for update.
I just ran it in our storybook.
Seems to be working without any errors.
got 530 accepted stories from 676.
the missing stories using Screener Steps so it makes sense...
one small thing that i noticed is that i got 60 new stories that should be already accepted.
other than that, looks great!

**Comment by sauce-gabriel** [comment]
Hello @igoldny; thank you for your valuable feedback.

I have been working with the team towards the support of the new Storybook, and happy to inform you and anyone affected by missing Screener Steps, that a new alpha version (`alpha-16`) was just released with fixes to the Screener Steps feature for those on **Storybook version 6.4.x or below.** 🎉  If you are using `screener-storybook@alpha`, reinstalling the dependencies should make the update for you.

One detail from your feedback that got my attention was the 60 new stories that were theoretically the same (should be accepted). This could be an impact of the puppeteer upgrade, so if you could share more details about it internally, it would be great. Puppeteer downgrade could be a thing if the impact is reasonably high.

**Comment by igoldny** [comment]
Amazing work everyone!
Tested `alpha-16` and everything is working including Screener Steps.

**Comment by sauce-tartar** [comment]
Heya @JBallin and @igoldny, we've released Screener Storybook 1.0 now with this feature added and crashing related to internal Storybook architecture shift mitigated.

Okay to close this?
